indra_gpt/processors/post_processors.py

```diff
@@ -0,0 +1,175 @@
+                    if isinstance(stmt_json_response, list):
+                        stmts_json = stmt_json_response
+                    else:
+                        stmts_json = [stmt_json_response]
+                # 3. Map the stmt class names to the correct class names 
+                stmts_json = [self.map_stmt_class_name(stmt_json) for stmt_json in stmts_json]
+                # 4. Remove fields with null values
+                stmts_json = [self.remove_empty_strings_and_lists(stmt_json) for stmt_json in stmts_json]
+                # 5. Update evidence since the generated statements likely have dummy values for evidence
+                stmts_json = [self.update_evidence(stmt_json, text, pmid, self.config) for stmt_json in stmts_json]
+                processed_stmts_json_list.append(stmts_json)
+            except Exception as e:
+                self.logger.error(f"Error processing response: {response} | Error: {e}")
+                processed_stmts_json_list.append({"error": str(e)})
+
+        preassembled_statements = [self.preassemble_pipeline(stmt) for stmt in processed_stmts_json_list]
+        self.logger.info(f"Completed post-processing: {len(preassembled_statements)} statements processed.")
+        return preassembled_statements
+    
+    def map_stmt_class_name(self, stmt_json):
+        stmt_classes = get_all_descendants(Statement)
+        stmt_mapping = {stmt_class.__name__.lower(): stmt_class.__name__ for stmt_class in stmt_classes}
+        try:
+            stmt_type = stmt_json["type"]
+            mapped_type = stmt_mapping.get(stmt_type.lower(), stmt_type)
+            stmt_json["type"] = mapped_type
+        except KeyError:
+            pass
+        return stmt_json
+    
+    def remove_empty_strings_and_lists(self, d):
+        for key, value in list(d.items()):
+            if isinstance(value, dict):
+                self.remove_empty_strings_and_lists(value)
+            elif isinstance(value, list):
+                for i in value:
+                    if isinstance(i, dict):
+                        self.remove_empty_strings_and_lists(i)
+            if value in [None, "", [], {}]:
+                del d[key]
+        return d
+    
+    def update_evidence(self, stmt_json, original_input_text, pmid, config):
+        # Define the actual evidence metadata
+        actual_evidence = {
+            "text": original_input_text,
+            "pmid": pmid,
+            "source_api": "indra_gpt",
+            "annotations": {"indra_gpt_config": config.base_config.__dict__}
+        }
+
+        # Check if "evidence" exists and is a list
+        evidence_list = stmt_json.get("evidence", [])
+
+        if evidence_list:
+            # Update existing evidence entries
+            for evidence in evidence_list:
+                evidence.update(actual_evidence)
+        else:
+            # If "evidence" is empty or missing, add the first evidence entry
+            stmt_json["evidence"] = [actual_evidence]
```

Comment:
Are there cases when there is more than one evidence already existing in the statements?